src/with-props.js

```diff
@@ -49,7 +49,7 @@ export function prop(definition: PropOptions | void): Function {
           normalised.serialize,
           val
         );
-        this._updateDebounced();
+        this.triggerUpdateBatched();
```

Comment:
This is now explicitly public. I feel that if people are going to author mixins they're going to need this stuff. For example, if someone were to have written `withState`.

Reply:
I'd leave only the async version for now, would want very good reasons to allow sync.

Reply:
I (we?) chatted to Rich Harris about this awhile back. Things that affect layout want sync. Unless we offer a way out from our default async / batched, there's going to be times when some want to force a sync render so consumers can inspect layout props.

Reply:
Since shadow DOM is an implementation detail, consumers shouldn't have an assumption of what it looks like at any given point, IMO. What they *should* rely on is the element's own properties/events, and those should be updated synchronously.

Reply:
What if you set a prop that causes a re-render. It "looks" different, and as a consumer you know that, and it's an implementation detail you have to be aware of, and now you want to recalc the offset height.

```js
elem.somePropThatRenders = true;

// Wrong.
elem.offsetHeight;

// Right.
Promise.resolve().then(() => elem.offsetHeight);
```

I don't think it's fair for them to need to call `triggerUpdate()` sync, but if there were a mixin that fixed this behaviour for async, they'll want to call the sync version inside a getter for the accessors to ensure the render queue is flushed.